packages/flutter_mini_sprite/lib/src/flutter_mini_sprite.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:flutter/material.dart';
+import 'package:mini_sprite/mini_sprite.dart';
+
+/// {@template mini_sprite_widget}
+///
+/// A widget that displays a [MiniSprite] on the screen.
+///
+/// {@endtemplate}
+class MiniSpriteWidget extends StatelessWidget {
+  /// {@macro mini_sprite_widget}
+  const MiniSpriteWidget({
+    super.key,
+    required this.sprite,
+    this.pixelSize = 1,
+    this.palette = const [Color(0xFFFFFFFF), Color(0xFF000000)],
+  });
+
+  /// The size of each pixel of the sprite, defaults to 1.
+  final double pixelSize;
+
+  /// The palette of the sprite, defaults to white and black.
+  final List<Color> palette;
+
+  /// The sprite to render.
+  final MiniSprite sprite;
+
+  @override
+  Widget build(BuildContext context) {
+    final width = sprite.pixels[0].length * pixelSize;
+    final height = sprite.pixels.length * pixelSize;
+
+    return SizedBox(
+      width: width,
+      height: height,
+      child: CustomPaint(
+        painter: _MiniSpritePainter(sprite, pixelSize, palette),
+      ),
+    );
+  }
+}
+
+class _MiniSpritePainter extends CustomPainter {
+  _MiniSpritePainter(this.sprite, this.pixelSize, this.palette);
+
+  final MiniSprite sprite;
+  final double pixelSize;
+  final List<Color> palette;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final paint = Paint();
+
+    for (var y = 0; y < sprite.pixels.length; y++) {
+      for (var x = 0; x < sprite.pixels[y].length; x++) {
+        if (sprite.pixels[y][x] != -1) {
```

Comment:
Are the X and Y axis flipped? 🤔

Reply:
I wouldn't consider it flipped because that is by design haha

I find it more natural to keep the vertical axis as the first dimension of the matrix, when projecting it to the plane, the positions are just as they are declared.

I am open to discussing this more if you feel otherwise though, but we should do that on a different PR :)